submissions/yhorodechna/frogger-oop/app.js

```diff
@@ -0,0 +1,156 @@
+            this.score = 0;
+            this.resetEnemiesSpeed();
+            alert(`You lose! Score: ${this.player.score}`);
+            this.player.resetPosition();
+        };
+    };
+};
+class Player {
+    constructor({ boardWidth, boardHeight, stepX, stepY, score }) {
+        this.sprite = 'images/char-boy.png';
+        this.boardWidth = boardWidth;
+        this.boardHeight = boardHeight;
+        this.stepX = stepX;
+        this.stepY = stepY;
+        this.height = 50;
+        this.width = 50;
+        this.score = score;
+        this.x = this.boardWidth - 3 * this.stepX;
+        this.y = this.boardHeight - 2.5 * this.stepY;
+    }
+    update = function (dt) {
+    };
+    render = function (dt) {
+        ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+    };
+    /* Method resetPosition reset player position */
+    resetPosition() {
+        this.x = this.boardWidth - 3 * this.stepX;
+        this.y = this.boardHeight - 2.5 * this.stepY;
+    };
+    /* Method handleInput depending on pressed key performs player actions (movements up, down,left,right )
+    when player moves up it is checked whether the player entered the water and if check passed the following actions take place :
+        player goes to the next level (score + 1)
+        the player is notified of his won,
+        is used method resetPosition witch reset player position */
+    handleInput = function (key) {
+        switch (key) {
+            case "up":
+                if (this.y > 0) this.y -= this.stepY;
+                if (this.y < 0) {
+                    setTimeout(() => {
+                        this.score++
+                        alert(`You won! Score:  ${this.score}`);
+                        this.resetPosition();
+                    }, 50)
+                };
+                break;
+            case "down":
+                if (this.y < this.boardHeight - 3 * this.stepY) this.y += this.stepY;
+                break;
+            case "left":
+                if (this.x > 0) this.x -= this.stepX;
+                break;
+            case "right":
+                if (this.x < this.boardWidth - this.stepX) this.x += this.stepX;
+                break;
+            default:
+                break;
+        };
+    };
+};
+const player = new Player({
+    boardWidth: CANVAS.width,
+    boardHeight: CANVAS.height,
+    boardWidth: CANVAS.width,
+    stepX: CANVAS.cellWidth,
+    stepY: CANVAS.cellHeight,
+    score: SCORE,
+});
+const allEnemies = [];
+for (let i = 0; i < 3; i++) {
+    allEnemies.push(new Enemy({
+        x: CANVAS.width - 6 * CANVAS.cellWidth,
+        y: CANVAS.height - (4.5 + i) * CANVAS.cellHeight,
+        speed: Math.floor(Math.random() * (100 - 40)) + 40,
+        boardWidth: CANVAS.width,
+        player: player,
+        width: CANVAS.cellWidth,
+        height: CANVAS.cellHeight,
+        resetEnemiesSpeed: RESET_ENEMIES_SPEED,
+    }));
+};
+console.log(allEnemies)
```

Comment:
Remove debugging and comments if there are any

Reply:
Deleted.